src/gluonts/dataset/multivariate_grouper.py

```diff
@@ -0,0 +1,191 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+# Standard library imports
+import logging
+import numpy as np
+import pandas as pd
+from typing import Callable
+
+# First-party imports
+from gluonts.dataset.common import ListDataset, DataEntry, Dataset
+from gluonts.core.component import validated
+from gluonts.dataset.field_names import FieldName
+
+
+class MultivariateGrouper:
+    """
+    The MultivariateGrouper takes a univariate dataset and groups it into a
+    single multivariate time series. Therefore, this class allows the user
+    to convert a univariate dataset into a multivariate dataset without making
+    a separate copy of the dataset.
+
+    The Multivariate Grouper has two different modes:
+    Training: For training data, the univariate time series get aligned to the
+       earliest time stamp in the dataset. Time series will be left and right
+       padded to produce an array of shape (dim, num_time_steps)
+    Test: The test dataset might have multiple start dates (usually because
+          the test dataset mimics a rolling evaluation scenario). In this case,
+          the univariate dataset will be split into n multivariate time series,
+          where n is the number of evaluation dates. Again, the
+          time series will be grouped but only left padded. Note that the
+          padded value will influence the prediction if the context length is
+          longer than the length of the time series.
+
+    Rules for padding for training and test datasets can be specified by the
+    user.
+    """
```

Comment:
please add annotations for constructor parameters here.

Reply:
Done.